### PR TITLE
Support enum unknown values

### LINKED ISSUE #58: Make enums loose
Today, it's always a breaking change to add a new option on a Enum on Artemis: the generated code will create an Enum based on the schema of on that code was generated, and if the server returns a new value on that enum, it will break with "ArgumentError".

To make enums retro-compatible, I suggest to (optionally) add an `UNKNOWN` value to every enum, and fallback to it on (well) unknown values.

### PR DESCRIPTION
**Pull request by leoiacovini**
This change aims to close https://github.com/comigor/artemis/issues/58 .

Adding a new value for each enum `ARTEMIS_UNKNOWN` and fall back to it when the JSON deserialization fails.

### CODE REVIEW

File: example/graphbrainz/lib/main.dart [hunk 1]
```diff
@@ -23,5 +21,5 @@ Future<void> main() async {
   final edSheeran = response.data.node as Artist;
   print(edSheeran.name);
   print(edSheeran.lifeSpan.begin);
-  print(edSheeran.spotify.href);
+//  print(edSheeran.spotify.href);
```
Comment (comigor) [inline example/graphbrainz/lib/main.dart]: Why was that commented?

File: lib/generator/data.dart [hunk 2]
```diff
@@ -31,20 +31,28 @@ class ClassProperty extends Equatable {
   /// Whether this parameter is required
   final bool isNonNull;
 
+  /// Whether this parameter corresponds to the __resolveType (or equivalent)
+  final bool isResolveType;
```
Comment (comigor) [inline lib/generator/data.dart]: you need to add this to `props` getter

File: lib/generator/print_helpers.dart [hunk 3]
```diff
@@ -113,8 +113,8 @@ Spec classDefinitionToSpec(ClassDefinition definition) {
       ..constructors.add(Constructor((b) {
         b
           ..optionalParameters.addAll(definition.properties
```
Comment (comigor) [inline lib/generator/print_helpers.dart]: those parameters on class constructors are actually only needed when we're generating input objects, but I'm fixing this behaviour on [visitor branch](https://github.com/comigor/artemis/tree/visitor)

**Review by comigor — CHANGES_REQUESTED** [review]
Just some small changes, but I still agree we need to have a way of enums not being breaking (and nobody uses extensions)

File: example/graphbrainz/lib/main.dart (hunk 1 above)
Comment (leoiacovini) [reply example/graphbrainz/lib/main.dart]: I was breaking with Null value 🤷‍♂ (probably something in the API changed)

Comment (comigor) [reply example/graphbrainz/lib/main.dart]: meh, just remove it then, please

**Review by comigor — APPROVED** [review]

**Comment by comigor** [comment]
Released as https://pub.dev/packages/artemis/versions/3.2.0

**Comment by PhilippS93** [comment]
How can I disable the option to generate the additional enum field?
E.g. when using `MyEnum.values` in code, everything breaks, when there is a custom enum constant, which ist not specified in the backend API.

Cheers

**Pull request merged**